### DISCUSSION TITLE
Google Summer of Code?

### DISCUSSION THREAD

**Discussion by kaveh808** [kaveh808/kons-9#182] [Ideas]
What do people think about applying to this program as a mentor organization? They do want projects with 1-2 years of track record, so we may be a bit early.

Curious if anyone has had experience with the program.

https://developers.google.com/open-source/gsoc/timeline

**Comment by awolven** [comment]
I don't know whether you consider me a slow or fast developer, but I tend
to think of those developers as way more gifted and efficient than me. I
think this is basically because I worked at Franz and I worked with a lot
of developers who were stars, like Gabor Melis and Andreas Fuchs.  But
maybe I just have the irrational fear of being an impostor.

**Reply by ghost** [reply]
Maybe if you sign-on as a mentor, you could prove yourself wrong.  Unfortunately some skills are less tangibly rewarded than others, but we all have our strengths.  Mentorship programs also help develop good managers as well as coders, which we sorely need.

**Reply by kaveh808** [reply]
For now I'll wait to hear from the BRL-CAD guys. Get some feedback at least.

**Reply by ghost** [reply]
That comment was actually directed at the "impostor syndrome" feeling, it does seem to apply to this topic in general.  I was just thinking of more motivation for doing it.  Keep in mind the BRL-CAD has some of the "amazing tool which is underappreciated" vibe like Common Lisp--and they get some great people to volunteer.  OK, I'll be patient now.

**Reply by kaveh808** [reply]
Oh, my bad. I misread the thread.

**Comment by ghost** [comment]
I haven't done it but I know being a mentor requires a time commitment, and I personally would only do it if I had a top person that would make it worth it, e.g. doing something I couldn't or way faster.  (The topic of general sweeps comes to mind in my case.)

I do know that the BRL-CAD people have been doing it a long time.  Here's a link to their participation in the program:

https://brlcad.org/wiki/Google_Summer_of_Code

When I was working with BRL-CAD, I know students made some significant contributions.  The maintainers are nice people, and you might ask them directly for their experiences, too.

**Reply by kaveh808** [reply]
What do you mean by general sweeps? I do some sweep operations in our system.

**Reply by ghost** [reply]
Here is a representative paper:

http://unknownroad.com/publications/SweepsTRApril2005.pdf

I was talking about it in another discussion about the use of it in CNC.  Also useful in interference detection of mechanical systems, autonomous vehicles and robotics.

**Reply by kaveh808** [reply]
Check out `demo-sweep-mesh.lisp`

**Reply by ghost** [reply]
> Check out demo-sweep-mesh.lisp

Nice, looks like you have the type of support for sweeps B-rep CAD systems typically use for design.

**Comment by ghost** [comment]
Looking at the BRL-CAD site just now, I saw that they were an umbrella organization for several other 3D CAx projects:

https://summerofcode.withgoogle.com/programs/2022/organizations/brl-cad

I also wasn't trying to be discouraging with my own attitude towards GSoC.  If you have time and/or an educational aptitude/goal (I'm not that good at teaching) that's great--just know it does require time on the part of the mentor.

**Reply by kaveh808** [reply]
Sounds like something worth investigating at least. Do you have an email for the BRL-CAD people responsible for GSoC?

**Reply by ghost** [reply]
From the above link: devs@brlcad.org

**Reply by ghost** [reply]
> Sounds like something worth investigating at least.

Especially in light of your background, and from the comments I've read here, the caliber of the other people working on this project you'll most likely be able to draw some great candidates.  I was talking to my brother about Jurassic Park and thought that if I was in a graphics program in college, participating in kons-9 would be quite an interesting opportunity if I had a chance to learn from someone who worked on that movie.

**Reply by kaveh808** [reply]
Thanks, I emailed them.

I like the idea of attracting students to the project. I suspect the biggest obstacle will be finding ones who are interested in CL. So many other more mainstream 3D apps out there.

I have had words of encouragement regarding the project from some Lisp elder statesmen, such as Peter Norvig and Scott Fahlman, which feels good.

**Comment by ghost** [comment]
CGAL participates in GSoC as well, e.g.:

https://github.com/CGAL/cgal/wiki/Project-Ideas#isosurfacing-2-marching-cubes-and-dual-contouring-in-octrees
